app/src/main/java/com/android/unio/ui/components/SearchPagerSection.kt

```diff
@@ -0,0 +1,169 @@
+package com.android.unio.ui.components
+
+import android.util.Log
+import androidx.compose.foundation.layout.*
+import androidx.compose.foundation.pager.*
+import androidx.compose.material3.*
+import androidx.compose.runtime.*
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.drawBehind
+import androidx.compose.ui.geometry.CornerRadius
+import androidx.compose.ui.geometry.Offset
+import androidx.compose.ui.geometry.Size
+import androidx.compose.ui.graphics.drawscope.Stroke
+import androidx.compose.ui.layout.onSizeChanged
+import androidx.compose.ui.unit.dp
+import com.android.unio.ui.theme.AppTypography
+import kotlinx.coroutines.launch
+
+/**
+ * A generalized composable for displaying a search bar with a sliding pager.
+ *
+ * @param T The type of the entity (e.g., Event, Member, Association).
+ * @param items The list of items to display in the pager.
+ * @param searchViewModel The view model handling the search logic.
+ * @param onItemSelected Callback when an item is selected from search results.
+ * @param cardContent A composable lambda that defines how to render each pager content.
+ * @param title The title displayed above the section.
+ * @param searchBar Composable function for rendering the search bar.
+ */
+@Composable
+fun <T> SearchPagerSection(
+    items: List<T>,
+    cardContent: @Composable (T) -> Unit,
+    searchBar: @Composable () -> Unit,
+    pagerState: PagerState
+) {
+  // Column layout to hold title, search bar, progress bar, and pager
+  Column(
+      horizontalAlignment = Alignment.CenterHorizontally,
+      modifier = Modifier.fillMaxWidth().padding(vertical = 8.dp)) {
+
+        // Sliding Progress Bar (if more than one item exists)
+        if (items.size > 1) {
+          // Search Bar Composable
+          //Box(modifier = Modifier.fillMaxWidth().padding(horizontal = 20.dp)) { searchBar() }
```

Comment:
As I have problem with the MemberSearchBar and I need the help of @Redd87, it will be in comments but I think it would be cool to implement it soon ! ; )